src/dependency-manager/src/utils/slugs.ts

```diff
@@ -0,0 +1,241 @@
+
+export type ArchitectSlug = string; // simply string that passes: ArchitectSlugValidator (ie "account-name")
+export type ComponentSlug = string; // "account-name/component-name"
+export type ComponentTag = string; // "tag"
+export type ComponentVersionSlug = string; // "account-name/component-name:tag"
+export type ServiceSlug = string; // "account-name/component-name/service-name"
+export type ServiceVersionSlug = string; // "account-name/component-name/service-name:tag"
+export type EnvironmentSlug = string; // "@environment-account-name/environment-name"
+export type ServiceInstanceSlug = string; // "account-name/component-name/service-name:tag@environment-account-name/environment-name"
```

Comment:
Note: I added support for a theoretical `ServiceInstance` slug which we don't yet use, but which I suspect we may find useful. It is globally unique (ie across environments), and is a friendly string that represents a deployed service including the environment.